### ISSUE TITLE
dev/build/lock.sh: Command not found

### ISSUE THREAD

**Issue by elliottslaughter** [swift-lang/swift-t#193]
I'm trying to build Swift/T from source as a part of Task Bench, and I'm hitting an error that looks like:

```
../../dev/build/lock.sh -c /home/runner/work/task-bench/task-bench/deps/swift/install/c-utils
make: ../../dev/build/lock.sh: Command not found
make: *** [Makefile:214: /home/runner/work/task-bench/task-bench/deps/swift/install/c-utils/install.txt] Error 127
```

You can see the build log as part of our CI here: https://github.com/StanfordLegion/task-bench/actions/runs/3041330396/jobs/4898338163

Here is how we are attempting to build:

https://github.com/StanfordLegion/task-bench/blob/29a2fd4f7e13615269fd0b34eb010b4a83863792/build_all.sh#L447-L478

This worked in Swift/T 1.4 with Ubuntu 18.04, but something bitrotted and an attempted upgrade to 1.5.0 to get rid of the bitrot has hit the `lock.sh` error. However, as best I can tell from reading through https://swift-lang.github.io/swift-t/guide.html#Build_configuration, this is more or less how Swift/T is intended to be built.

I'd appreciate any pointers on what we're getting wrong.

**Comment by j-woz** [comment]
Sorry to hear this- does dev/build/lock.sh exist after you unpack the TGZ?

**Comment by elliottslaughter** [comment]
```
$ wget -nv http://swift-lang.github.io/swift-t-downloads/1.5/swift-t-1.5.0.tar.gz
2022-09-13 12:25:15 URL:https://swift-lang.github.io/swift-t-downloads/1.5/swift-t-1.5.0.tar.gz [8647943/8647943] -> "swift-t-1.5.0.tar.gz" [1]
$ tar xf swift-t-1.5.0.tar.gz 
$ find swift-t-1.5.0 -name 'lock.sh'
swift-t-1.5.0/dev/build/lock.sh
```

It's in the tarball. So I guess I need to check how it's managing to go missing (or maybe we have the wrong path or something) in the build.

**Comment by elliottslaughter** [comment]
There is an initial fix in #196. The issue was hard-coded paths for `/bin/zsh`.

Now I'm seeing this in the Turbine configure:

```
+ ./configure --config-cache --prefix=/task-bench/deps/swift/install/turbine --with-c-utils=/task-bench/deps/swift/install/c-utils --with-adlb=/task-bench/deps/swift/install/lb --with-tclsh-local=/task-bench/deps/swift/install/bin/tclsh8.6 --with-tcl-lib-dir=/task-bench/deps/swift/install/lib --with-tcl-include=/task-bench/deps/swift/install/include --with-tcl-syslib-dir=/task-bench/deps/swift/install/lib --enable-checkpoint=no --disable-static-pkg --without-zlib --disable-checkpoint --with-hdf5=no --disable-static
configure: WARNING: unrecognized options: --with-tcl-syslib-dir
configure: creating cache config.cache
Turbine version: 1.3.0
...
configure: Tcl version is not specified.
configure: Tcl location not specified, trying to find tclsh on path
checking for tclsh8.6 on path... yes: /task-bench/deps/swift/install/bin/tclsh8.6
Found Tcl at /task-bench/deps/swift/install
using Tcl lib dir: /task-bench/deps/swift/install/lib
checking for local Tcl (version: 0) executable in: /task-bench/deps/swift/install/bin/tclsh8.6... 
Could not find Tcl directory: /task-bench/deps/swift/install/bin/tclsh8.6
configure: error: failed!
```

I suppose I must be missing something in the way we inform the build about the location of TCL, but it is confusing because the configure script clearly seems to have found the binary.

**Comment by elliottslaughter** [comment]
I seem to have a working solution in https://github.com/StanfordLegion/task-bench/pull/89.

I think this means that once that passes, #196 is sufficient to resolve this.
